### DISCUSSION TITLE
Selecting which page in a multipage pdf gets processed in file conversions

### DISCUSSION THREAD

**Discussion by threeaccents** [libvips/libvips#2126] [Q&A]
Currently, for multipage PDFs being converted to another format only the first page of the PDF gets converted. I was wondering if there is the possibility to tell vips which page of the multipage pdf to convert. So if I have a 10-page pdf instead of page 1 getting converted I could supply a flag to tell it to convert page 8 instead.

**Comment by jcupitt** [comment]
Hello @threeaccents,

Sure, just set `page`. For example:

```
vips copy Audi_US\ R8_2017-2.pdf[page=4] x.png
```

To convert page 4. Pages number from zero.

You can set a number of pages to convert with `n`, or use -1 to mean all pages. For example:

```
vips copy Audi_US\ R8_2017-2.pdf[n=-1] x.tif
```

Will write a multipage tiff with every page in the PDF as a separate page in the TIFF.

Docs here:

https://libvips.github.io/libvips/API/current/VipsForeignSave.html#vips-pdfload

**Reply by threeaccents** [reply]
@jcupitt Thank you! 

P.S. wasn't aware of GitHub discussions this is awesome!

**Reply by jet10000** [reply]
> 

vips copy abc.pdf[page=3] a.jpg     
zsh: no matches found: [page=3]

**Reply by jcupitt** [reply]
I think zsh has a special meaning for the `[]` characters. Maybe try:

```
vips copy "abc.pdf[page=3]" a.jpg
```

**Comment by jet10000** [comment]
```
vips pdfload abc.pdf a.jpg --n=6 
```
